src/source/Ice/IceAgent.c

```diff
@@ -8,15 +8,16 @@ extern StateMachineState ICE_AGENT_STATE_MACHINE_STATES[];
 extern UINT32 ICE_AGENT_STATE_MACHINE_STATE_COUNT;
 
 STATUS createIceAgent(PCHAR username, PCHAR password, UINT64 customData, PIceAgentCallbacks pIceAgentCallbacks,
-                      PRtcConfiguration pRtcConfiguration, TIMER_QUEUE_HANDLE timerQueueHandle, PIceAgent* ppIceAgent)
+                      PRtcConfiguration pRtcConfiguration, TIMER_QUEUE_HANDLE timerQueueHandle,
+                      PConnectionListener pConnectionListener, PIceAgent* ppIceAgent)
 {
     ENTERS();
 
     STATUS retStatus = STATUS_SUCCESS;
     PIceAgent pIceAgent = NULL;
     UINT32 i, turnServerCount = 0;
 
-    CHK(ppIceAgent != NULL && username != NULL && password != NULL, STATUS_NULL_ARG);
+    CHK(ppIceAgent != NULL && username != NULL && password != NULL && pConnectionListener != NULL, STATUS_NULL_ARG);
```

Comment:
Do we need the pConnectionListener to be mandatory?

Reply:
yes, it's the network listener thread.